admin/install.php

```diff
@@ -1121,6 +1124,30 @@ function_exists( 'mysqli_stmt_get_result' ),
 				$t_operation = $g_upgrade[$i][0];
 				$t_target = $g_upgrade[$i][1][0];
 
+				# SQLite: Patch schema
+				if( $f_db_type == 'sqlite3' && isset( $g_upgrade[$i][1][1] ) ) {
+					$t_request = $g_upgrade[$i][1][1];
+
+					# Rely on auto-detection
+					$t_request = str_replace( [ 'UNSIGNED', 'AUTOINCREMENT' ], '', $t_request );
+
+					# Remove the second PRIMARY
+					$t_request = str_replace( 'PRIMARY KEY', 'PRIMARY', $t_request );
+					while( substr_count( $t_request, 'PRIMARY' ) > 1 ) {
+						$t_pos = strrpos( $t_request, 'PRIMARY' );
+						$t_request = substr( $t_request, 0, $t_pos ) . substr( $t_request, $t_pos + 7 );
+					}
+					$t_request = str_replace( 'PRIMARY', 'PRIMARY KEY', $t_request );
+
+					# Remove NOT NULL of BLOB, use db_update_blob() instead
+					$t_request = preg_replace( '/B\s+NOTNULL/', 'B', $t_request );
+
+					# Replace VARCHAR(x) with TEXT
+					$t_request = preg_replace( '/C\([0-9]+\)/', 'X', $t_request );
```

Comment:
Is this really needed ? I believe SQLite will just swallow whatever type it's given and use TEXT internally as appropriate.

Reply:
The "B NOTNULL" i.e. "BLOB NOT NULL" has to be changed to just "BLOB" because SQLite handles blob separately in the two SQL calls, and in the first call it's actually NULL.

The "C(x)" i.e. VARCHAR(x) needs to be changed to "TEXT" to mitigate the missed AlterColumnSQL which is commonly used to expand the column size.

Reply:
Actually VARCHAR doesn't need to be touched, SQLite doesn't take advantage of the restrictions as I thought. And BLOB NOT NULL I redesigned as BLOB NOT NULL DEFAULT ‘’ and put it directly into schema.php.